hermes/crates/cbork/cddl-parser/src/lib.rs

```diff
@@ -45,41 +78,61 @@ pub const POSTLUDE: &str = include_str!("grammar/postlude.cddl");
 /// # Examples
 ///
 /// ```rs
-/// use cddl_parser::parse_cddl;
+/// use cddl_parser::{parse_cddl, Extension};
 /// use std:fs;
 ///
 /// let input = fs::read_to_string("path/to/your/file.cddl").unwrap();
-/// let result = parse_cddl(&input);
+/// let result = parse_cddl(&input, &Extension::CDDLParser);
 /// assert!(result.is_ok());
 /// ```
-pub fn parse_cddl(input: &str) -> Result<(), Box<CDDLError>> {
-    let result = CDDLParser::parse(Rule::cddl, input);
-
-    match result {
-        Ok(c) => println!("{c:?}"),
-        Err(e) => {
-            println!("{e:?}");
-            println!("{e}");
-            return Err(Box::new(CDDLError::from(e)));
+pub fn parse_cddl(input: &str, extension: &Extension) -> Result<(), Box<CDDLError>> {
+    match extension {
+        Extension::RFC8610Parser => {
+            let result = rfc_8610::RFC8610Parser::parse(rfc_8610::Rule::cddl, input);
+
+            match result {
```

Comment:
This is common logic,  just have the `rfc_8610::...` use `map_err` https://doc.rust-lang.org/std/result/enum.Result.html#method.map_err to convert the error to the enum error type.
Then the logic is cleaner.
You can then use:
```
let result = match extension {
...
}
```

and then just print the error in one after the match.